mkdocs/commands/serve.py

```diff
@@ -1,21 +1,49 @@
 from __future__ import unicode_literals
+
 import logging
 import shutil
 import tempfile
 
+from os.path import isfile, join
 from mkdocs.commands.build import build
 from mkdocs.config import load_config
 
 log = logging.getLogger(__name__)
 
 
+def _get_handler(site_dir):
+
+    from tornado.web import StaticFileHandler
+    from tornado.template import Loader
+
+    class WebHandler(StaticFileHandler):
+
+        def write_error(self, status_code, **kwargs):
+
+            if status_code in (404, 500):
+                error_page = '{}.html'.format(status_code)
+                if isfile(join(site_dir, error_page)):
+                    self.write(Loader(site_dir).load(error_page).generate())
+                else:
+                    super(WebHandler, self).write_error(status_code, **kwargs)
+
+    return WebHandler
+
+
 def _livereload(host, port, config, builder, site_dir):
 
     # We are importing here for anyone that has issues with livereload. Even if
     # this fails, the --no-livereload alternative should still work.
     from livereload import Server
 
-    server = Server()
+    class LiveReloadServer(Server):
+
+        def get_web_handlers(self, script):
+            handlers = super(LiveReloadServer, self).get_web_handlers(script)
+            handlers = handlers.pop()
+            return [(handlers[0], _get_handler(site_dir), handlers[2],)]
```

Comment:
And on closer inspection I realize that the method returns a list of one tuple which is clearly the format returned by the method of the parent class. Its all clear enough. 

Although I wonder if this would be better:

```python
handlers = super(LiveReloadServer, self).get_web_handlers(script)
return [(handlers[0][0], _get_handler(site_dir), handlers[0][2],)]
```